### PR TITLE
feat: typings for clients

### PR DESCRIPTION
**Pull request by lkhari**
## What did you implement:
Added typings to the most of the clients in the powertool libraries

## How did you implement it:
Added a index.d.ts file

## How can we verify it:
Create a typescript project import any of the clients and see expect typings

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [ ] Provide verification config / projects / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by derevandal** [comment]
Awesome!
I'm looking for this!

When this PR will be merged?

Thanks

File: packages/lambda-powertools-cloudwatchevents-client/package.json [hunk 1]
```diff
@@ -21,4 +22,4 @@
   "publishConfig": {
     "access": "public"
   }
-}
+}
```
Comment (simontabor) [inline packages/lambda-powertools-cloudwatchevents-client/package.json:25]: Nitpicky I know but could we get the newlines re-added? (a few repeated examples in this PR)
```suggestion
}

```

Comment (lkhari) [reply packages/lambda-powertools-cloudwatchevents-client/package.json]: All done!

**Comment by lkhari** [comment]
Any updates on these merges?

**Pull request merged**